internal/store/interfaces.go

```diff
@@ -27,7 +28,13 @@ type Gateway interface {
 	Bind(ctx context.Context, route Route) []string
 	Remove(ctx context.Context, id string) error
 	Resolve() core.ResolvedGateway
-	CanFetchSecrets(ctx context.Context, secrets []string) (bool, error)
+	CanFetchSecrets(secrets []string) (bool, error)
+}
+
+type NewGateway interface {
```

Comment:
This is the new simplified interface for the gateway which `K8sGateway` already fulfills. It will fully replace the `Gateway` interface above in a followup PR